### PR TITLE
Improved log formatting to reduce splunk spam

### PR DESCRIPTION
**Pull request by ri-ch**

### CODE REVIEW

**Review by mfjarvis — DISMISSED** [review]

File: coordinator/src/app.ts [hunk 1]
```diff
@@ -2,11 +2,12 @@ import {Boom} from "@hapi/boom"
...
 import routes from "./routes"
 import {toOperationOutcome, FhirMessageProcessingError} from "./models/errors/processing-errors"
+import HapiPino from "hapi-pino"
 
 const preResponse = function (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit) {
   const response = request.response
   if (response instanceof Boom) {
-    console.log(response)
+    request.log("info", response)
     if (response instanceof FhirMessageProcessingError) {
```
Comment (mfjarvis) [inline coordinator/src/app.ts:11]: Might be nice to log FhirMessageProcessingError at info level and everything else at error level?

Comment (ri-ch) [reply coordinator/src/app.ts:11]: Done :+1:

**Review by conner-jowitt — DISMISSED** [review]

File: coordinator/tests/messagehandling.spec.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import {Boom, boomify} from "@hapi/boom"
```
Comment (mfjarvis) [inline coordinator/tests/messagehandling.spec.ts:1]: Is this file testing any of our code or is it basically testing `boomify`?

File: coordinator/src/app.ts [hunk 3]
```diff
@@ -2,14 +2,15 @@ import {Boom} from "@hapi/boom"
 import Hapi from "@hapi/hapi"
 import routes from "./routes"
 import {toOperationOutcome, FhirMessageProcessingError} from "./models/errors/processing-errors"
+import HapiPino from "hapi-pino"
 
 const preResponse = function (request: Hapi.Request, responseToolkit: Hapi.ResponseToolkit) {
   const response = request.response
   if (response instanceof Boom) {
```
Comment (mfjarvis) [inline coordinator/src/app.ts:9]: I think you need to swap these conditions round and handle `FhirMessageProcessingError` first

Comment (mfjarvis) [reply coordinator/src/app.ts:9]: Double checking now

File: coordinator/tests/messagehandling.spec.ts (hunk 2 above)
Comment (ri-ch) [reply coordinator/tests/messagehandling.spec.ts:1]: Good spot. Ill get rid of this

File: coordinator/src/app.ts (hunk 3 above)
Comment (mfjarvis) [reply coordinator/src/app.ts:9]: Yeah, looks like we're now returning 500s for validation errors where we used to return 400s

Comment (mfjarvis) [reply coordinator/src/app.ts]: Pushed a fix

**Review by mfjarvis — APPROVED** [review]

File: coordinator/src/app.ts [hunk 4]
```diff
@@ -29,8 +30,19 @@ const init = async () => {
...
+      prettyPrint: process.env.ENVIRONMENT_NAME === "local",
+      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
+      redact: ["req.headers.authorization"]
+    }
+  })
+
   await server.start()
-  console.log("Server running on %s", server.info.uri)
+  server.log("info", `Server running on ${server.info.uri}`)
+  server.logger
```
Comment (mfjarvis) [inline coordinator/src/app.ts:45]: ?

Comment (mfjarvis) [reply coordinator/src/app.ts:45]: Assume this was left over from debugging but cba waiting for pipelines to run again

**Pull request merged**